transport/awslambda/encode_decode.go

```diff
@@ -0,0 +1,16 @@
+package awslambda
+
+import (
+	"context"
+)
+
+// DecodeRequestFunc extracts a user-domain request object from an
+// AWS lambda's payload.
+type DecodeRequestFunc func(context.Context, []byte) (interface{}, error)
+
+// EncodeResponseFunc encodes the passed response object into []byte,
+// ready to be sent as AWS lambda response.
+type EncodeResponseFunc func(context.Context, interface{}) ([]byte, error)
+
+// ErrorEncoder is responsible for encoding an error.
```

Comment:
To what? Is the []byte returned just like a response?

Reply:
are u referring to `ErrorEncoder` ? if it is - then yes, the []byte is for controlling the response, for example in the context of building an API, we might want to response gracefully with a body of JSON describing detail of the error to Consumer.